state-chain/pallets/cf-threshold-signature/src/lib.rs

```diff
@@ -1168,31 +1156,41 @@ pub mod pallet {
 			)
 		}
 
-		#[pallet::call_index(7)]
-		#[pallet::weight(T::Weights::set_keygen_response_timeout())]
-		pub fn set_keygen_response_timeout(
+		/// Apply a list of configuration updates to the pallet.
+		///
+		/// Requires Governance.
+		#[pallet::call_index(9)]
+		#[pallet::weight(T::Weights::update_pallet_config())]
+		pub fn update_pallet_config(
 			origin: OriginFor<T>,
-			new_timeout: BlockNumberFor<T>,
+			update: PalletConfigUpdate,
 		) -> DispatchResult {
 			T::EnsureGovernance::ensure_origin(origin)?;
 
-			if new_timeout != KeygenResponseTimeout::<T, I>::get() {
-				KeygenResponseTimeout::<T, I>::put(new_timeout);
-				Pallet::<T, I>::deposit_event(Event::KeygenResponseTimeoutUpdated { new_timeout });
-			}
-
-			Ok(())
-		}
+			match update {
+				PalletConfigUpdate::ThresholdSignatureResponseTimeout { new_timeout } => {
+					let new_timeout: BlockNumberFor<T> = new_timeout.into();
+					if new_timeout != ThresholdSignatureResponseTimeout::<T, I>::get() {
```

Comment:
I'm not sure we need this check: it just adds additional noise (and possibly confusion: 'why was my update not applied?'). We can assume that governance are competent enough not to make this kind of mistake.

Reply:
This was the previous behaviour, I would be happy to make it work like you suggest.